src/1_utils.js

```diff
@@ -20,6 +20,76 @@ utils.retries = 2; // Value specifying the number of times that a Branch API cal
 utils.retry_delay = 200; // Amount of time in milliseconds to wait before re-attempting a timed-out request to the Branch API.
 utils.timeout = 5000; // Duration in milliseconds that the system should wait for a response before considering any Branch API call to have timed out.
 
+utils.userPreferences = {
+	trackingDisabled: false,
+	whiteListedEndpointsWithData: {
+		'/v1/open': { 'link_identifier':'\\d+' },
+		'/v1/event': { 'event': 'pageview' },
+		'/v1/branchview': {}
+	},
+	allowErrorsInCallback: false
+};
+
+// Used by 3_api.js to determine whether a request should be blocked
+utils.userPreferences.shouldBlockRequest = function(url, requestData) {
```

Comment:
I think it might be more clear to just declare this within the object above. I was a little confused when I saw this method used and looked for it in the object above and didn't see it.